webapp/src/pages/eventConfig/components/EventConfigComponent.js

```diff
@@ -31,23 +69,37 @@ class EventConfigComponent extends Component {
     }
   }
 
-  onClickCancel = () => {
-    this.setState({
-      updatedEvent: this.state.preEvent,
-      hasBeenUpdated: false
-    });
-  };
-
   onClickSubmit = () => {
-    // PUT
-    eventService.update(this.state.updatedEvent).then(result => {
-      this.setState({
-        preEvent: result.event,
-        updatedEvent: result.event,
-        hasBeenUpdated: false,
-        error: result.error
+    const isValid = this.validateEventDetails();
+    if (isValid) {
+      if (this.state.isNewEvent) { // POST
+        eventService.create(this.state.updatedEvent).then(result => {
+          this.setState({
+            preEvent: result.event,
+            updatedEvent: result.event,
+            hasBeenUpdated: false,
+            error: result.error
+          });
+        });
+      }
+      else { // PUT
+        eventService.update(this.state.updatedEvent).then(result => {
+          this.setState({
+          preEvent: result.event,
+          updatedEvent: result.event,
+          hasBeenUpdated: false,
+          error: result.error
+        });
       });
-    });
+      }
+    }
+    else {
+      console.log('invalid');
+      //TODO: this doesn't work - errors not showing
+      return <div>
+        {this.state.errors && this.state.showErrors && this.getErrorMessages(this.state.errors)}
+        </div>
```

Comment:
You should move this block into your render function, maybe just above your submit button, then it will automatically render when this.state.errors becomes non-empty and this.state.showErrors becomes true.

Reply:
Done